benchmarkcommon/src/test/java/com/ss/benchmark/httpclient/HCPerformanceTests.java

```diff
@@ -0,0 +1,395 @@
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * @author sharath.srinivasa
+ */
+@Test(groups = "performance")
+public abstract class HCPerformanceTests {
+
+    protected static final String ECHO_DELAY_BASE_URL = "/echodelayserv/echo";
+    protected static final String ECHO_DELAY_SHORT_URL = "/echodelayserv/echo/short";
+    protected static final String ECHO_DELAY_LONG_URL = "/echodelayserv/echo/long";
+    protected static final String BASE_URL = "http://localhost:8080";
+
+    protected static final int EXECUTIONS = 10_000;
+
+
+    private static final Logger logger = LoggerFactory.getLogger(HCPerformanceTests.class);
+
+    protected final MetricRegistry metricRegistry = new MetricRegistry();
+    protected final ScheduledReporter reporter = ConsoleReporter.forRegistry(metricRegistry).convertDurationsTo(TimeUnit.MILLISECONDS).build();
+
+
+    private Set<CountDownLatch> latches = new HashSet<>();
+
+    private HC client;
+
+    @BeforeTest
+    public void initializeTest() {
+        reporter.start(1, TimeUnit.HOURS);
+
+        client = getClient();
+
+        client.setup(BASE_URL);
+
+        // this is simply to warmup the connection pool
+        logger.info("Start warmup");
+        for (int i = 0; i < EXECUTIONS; i++) {
+            syncGET(metricRegistry.timer(MetricRegistry.name(this.getClass(), "warmup", "timing")),
+                    metricRegistry.counter(MetricRegistry.name(this.getClass(), "warmup", "errorRate")));
+        }
+        logger.info("Completed warmup");
+    }
+
+    @AfterTest
+    public void finalizeTest() {
+        for (CountDownLatch latcher : latches) {
+            try {
+                latcher.await();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+
+        reporter.report();
+        reporter.close();
+    }
+
+    @Test(groups = {"sync", "blocking"})
+    public void testSyncBlockingShortGET() {
+        String method = myName();
+        logger.info("Start " + method);
+
+        for (int i = 0; i < EXECUTIONS; i++) {
+            syncGET(metricRegistry.timer(MetricRegistry.name(this.getClass(), method, "timing")),
+                    metricRegistry.counter(MetricRegistry.name(this.getClass(), method, "errorRate")));
+        }
+
+        logger.info("Completed " + method);
+    }
+
+    @Test(invocationCount = EXECUTIONS, threadPoolSize = 100, groups = {"async", "blocking"})
+    public void testAsyncBlockingShortGET() {
+        String method = myName();
+
+        logger.info("Start " + method);
+
+        syncGET(metricRegistry.timer(MetricRegistry.name(this.getClass(), method, "timing")),
```

Comment:
It's confusing to see `syncGET` used here and `syncPost` used in a test named `testSyncBlocking..`.  Even so, it's probably just me.  I don't want to get too caught up here.  If it makes sense to you, just ignore this.  I'll see how I feel if I add the Apache Async client.